azurerm/data_source_network_security_group.go

```diff
@@ -21,8 +24,17 @@ func dataSourceArmNetworkSecurityGroup() *schema.Resource {
 			"location": locationForDataSourceSchema(),
 
 			"security_rule": {
-				Type:     schema.TypeList,
+				Type:     schema.TypeSet,
```

Comment:
The datasource should be a `TypeList` because a set is difficult to address:
list `security_rule.0.name ` vs set `security_rule.1400690338.name `